ci-operator/step-registry/ibmcloud/provision/kms-key/ibmcloud-provision-kms-key-commands.sh

```diff
@@ -42,31 +42,62 @@ cat >${key_file} <<EOF
 }
 EOF
 
-instance_name="${CLUSTER_NAME}-kp"
-ibmcloud resource service-instance-create ${instance_name} kms tiered-pricing ${region}
 
-id=$(ibmcloud resource service-instance ${instance_name} --id -q | awk '{print $2}')
-echo "$(jq --arg idarg "$id" '. += {"id": $idarg}' $key_file)" > $key_file
+instance_m_name="${CLUSTER_NAME}-kpm"
+instance_w_name="${CLUSTER_NAME}-kpw"
 
-export KP_INSTANCE_ID=${id}
+ibmcloud resource service-instance-create ${instance_m_name} kms tiered-pricing ${region}
+id_m=$(ibmcloud resource service-instance ${instance_m_name} --id -q | awk '{print $2}')
+echo "$(jq --arg idarg "$id_m" '. += {"id_master": $idarg}' $key_file)" > $key_file
+
+ibmcloud resource service-instance-create ${instance_w_name} kms tiered-pricing ${region}
+id_w=$(ibmcloud resource service-instance ${instance_w_name} --id -q | awk '{print $2}')
+echo "$(jq --arg idarg "$id_w" '. += {"id_worker": $idarg}' $key_file)" > $key_file
 
 key_meterial=$(openssl rand -base64 32)
-key_name="${instance_name}-key"
+
+key_name_m="${instance_m_name}-key"
+key_name_w="${instance_w_name}-key"
+
 tmpFile=$(mktemp)
-ibmcloud kp key create ${key_name} -k ${key_meterial} -o json > ${tmpFile}
+ibmcloud kp key create ${key_name_m} -k ${key_meterial} -i ${id_m} -o json > ${tmpFile}
+
+keyCRN_m=$(jq -r .crn $tmpFile)
+
+keyID_m=$(jq -r .id $tmpFile)
+echo "$(jq --arg idarg "$keyID_m" '. += {"keyID_master": $idarg}' $key_file)" > $key_file
 
-keyCRN=$(jq -r .crn $tmpFile)
+ibmcloud kp key create ${key_name_w} -k ${key_meterial} -i ${id_w} -o json > ${tmpFile}
 
-keyID=$(jq -r .id $tmpFile)
-echo "$(jq --arg idarg "$keyID" '. += {"keyID": $idarg}' $key_file)" > $key_file
+keyCRN_w=$(jq -r .crn $tmpFile)
 
+keyID_w=$(jq -r .id $tmpFile)
+echo "$(jq --arg idarg "$keyID_w" '. += {"keyID_worker": $idarg}' $key_file)" > $key_file
+
+echo "dump ${key_file}..."
 cat ${key_file}
 
+# cat > "${SHARED_DIR}/ibm_kpKey.yaml" << EOF
+# platform:
+#   ibmcloud:
+#     defaultMachinePlatform:
+#       bootVolume:
+#         encryptionKey: "${keyCRN}"
+#     resourceGroupName: ${resource_group}
+# EOF
+
 cat > "${SHARED_DIR}/ibm_kpKey.yaml" << EOF
+controlPlane:
```

Comment:
I suggest to the step should support the kms key setting for defaultMachinePlatform, compute and master separately, the priority is compute | master should be higher than  defaultMachinePlatform when all of them are set, a good example is something like `ipi-conf-azure-osdisk-disksize`

Reply:
I will add IBMCLOUD_CONTROL_PLANE_ENCRYPTIONKEY  IBMCLOUD_COMPUTE_ENCRYPTIONKEY and IBMCLOUD_DEFAULT_MACHINE_ENCRYPTIONKEY  enviroment value

Reply:
ENCRYPTIONKEY - > ENCRYPTION_KEY